### PR TITLE
WT-5001 Migrate jenkins wiredtiger-test-format-stress-sanitizer-ppc

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
Evergreen Patch build [link](https://evergreen.mongodb.com/version/5e45de313e8e86781c2bd9db)

### CODE REVIEW

**Comment by raviprakashgiri29** [comment]
@lukech CC is set to the system default "clang" binary here as a workaround. I have put a comment to change it back to mongodbtoolchain "clang" binary after BUILD-10248 fix.

File: test/evergreen.yml [hunk 1]
```diff
@@ -2136,6 +2136,40 @@ tasks:
...
+      - func: "compile wiredtiger"
+        vars:
+          # CC is set to the system default "clang" binary here as a workaround. Change it back to mongodbtoolchain "clang" binary after BUILD-10248 fix.
+          configure_env_vars: CCAS=/opt/mongodbtoolchain/v3/bin/gcc CC=/usr/bin/clang CXX=/opt/mongodbtoolchain/v3/bin/clang++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fsanitize=address -fno-omit-frame-pointer -I/opt/mongodbtoolchain/v3/lib/gcc/ppc64le-mongodb-linux/8.2.0/include"
+          posix_configure_flags: --enable-diagnostic --with-builtins=lz4,snappy,zlib
+      - func: "format test script"
+        vars:
+          test_env_vars: ASAN_OPTIONS="abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=`ls /usr/bin/llvm-symbolizer* | tail -1`
+          # run for 2 hours ( 2 * 60 = 120 minutes), don't stop at failed tests, use default config
+          format_test_script_args: -t 120
```
Comment (lukech) [inline test/evergreen.yml]: Is there any special reason to set the run time to 2 hrs? The Jenkins job triggers itself continuously.

File: test/evergreen.yml [hunk 2]
```diff
@@ -2136,6 +2136,40 @@ tasks:
...
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          # CC is set to the system default "clang" binary here as a workaround. Change it back to mongodbtoolchain "clang" binary after BUILD-10248 fix.
+          configure_env_vars: CCAS=/opt/mongodbtoolchain/v3/bin/gcc CC=/usr/bin/clang CXX=/opt/mongodbtoolchain/v3/bin/clang++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fsanitize=address -fno-omit-frame-pointer -I/opt/mongodbtoolchain/v3/lib/gcc/ppc64le-mongodb-linux/8.2.0/include"
+          posix_configure_flags: --enable-diagnostic --with-builtins=lz4,snappy,zlib
+      - func: "format test script"
+        vars:
+          test_env_vars: ASAN_OPTIONS="abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=`ls /usr/bin/llvm-symbolizer* | tail -1`
```
Comment (lukech) [inline test/evergreen.yml]: Can we set ASAN_OPTIONS the same way as task `format-stress-sanitizer-test`?

File: test/evergreen.yml [hunk 3]
```diff
@@ -2136,6 +2136,40 @@ tasks:
...
 
+  - name: format-stress-sanitizer-test-ppc
+    # Set 2.5 hours timeout (60 * 60 * 2.5)
+    exec_timeout_secs: 9000
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          # CC is set to the system default "clang" binary here as a workaround. Change it back to mongodbtoolchain "clang" binary after BUILD-10248 fix.
+          configure_env_vars: CCAS=/opt/mongodbtoolchain/v3/bin/gcc CC=/usr/bin/clang CXX=/opt/mongodbtoolchain/v3/bin/clang++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fsanitize=address -fno-omit-frame-pointer -I/opt/mongodbtoolchain/v3/lib/gcc/ppc64le-mongodb-linux/8.2.0/include"
```
Comment (lukech) [inline test/evergreen.yml]: Not sure if CCAS setting makes any difference. Suggest to use the same `configure_env_vars` setting as function `compile wiredtiger address sanitizer`.

File: test/evergreen.yml [hunk 4]
```diff
@@ -2136,6 +2136,40 @@ tasks:
...
+
+  - name: format-stress-sanitizer-smoke-test-ppc
+    # Set 7 hours timeout (60 * 60 * 7)
+    exec_timeout_secs: 25200
+    commands:
+      - func: "get project"
+      - func: "compile wiredtiger"
+        vars:
+          # CC is set to the system default "clang" binary here as a workaround. Change it back to mongodbtoolchain "clang" binary after BUILD-10248 fix.
+          configure_env_vars: CCAS=/opt/mongodbtoolchain/v3/bin/gcc CC=/usr/bin/clang PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fsanitize=address -fno-omit-frame-pointer -I/opt/mongodbtoolchain/v3/lib/gcc/ppc64le-mongodb-linux/8.2.0/include"
```
Comment (lukech) [inline test/evergreen.yml]: Ditto.

File: test/evergreen.yml [hunk 5]
```diff
@@ -2136,6 +2136,40 @@ tasks:
...
+          posix_configure_flags: --enable-diagnostic --with-builtins=lz4,snappy,zlib         
+      - func: "format test script"
+        # to emulate the original Jenkins job's test coverage, we are running the smoke test 16 times
+        # run smoke tests, don't stop at failed tests, use default config
+        vars:
+          test_env_vars: ASAN_OPTIONS="abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=`ls /usr/bin/llvm-symbolizer* | tail -1`
```
Comment (lukech) [inline test/evergreen.yml]: Ditto.

File: test/evergreen.yml [hunk 6]
```diff
@@ -2136,6 +2136,40 @@ tasks:
...
+          # CC is set to the system default "clang" binary here as a workaround. Change it back to mongodbtoolchain "clang" binary after BUILD-10248 fix.
+          configure_env_vars: CCAS=/opt/mongodbtoolchain/v3/bin/gcc CC=/usr/bin/clang CXX=/opt/mongodbtoolchain/v3/bin/clang++ PATH=/opt/mongodbtoolchain/v3/bin:$PATH CFLAGS="-ggdb -fPIC -fsanitize=address -fno-omit-frame-pointer -I/opt/mongodbtoolchain/v3/lib/gcc/ppc64le-mongodb-linux/8.2.0/include"
+          posix_configure_flags: --enable-diagnostic --with-builtins=lz4,snappy,zlib
+      - func: "format test script"
+        vars:
+          test_env_vars: ASAN_OPTIONS="abort_on_error=1:disable_coredump=0" ASAN_SYMBOLIZER_PATH=`ls /usr/bin/llvm-symbolizer* | tail -1`
+          # run for 2 hours ( 2 * 60 = 120 minutes), don't stop at failed tests, use default config
+          format_test_script_args: -t 120
+
+  - name: format-stress-sanitizer-smoke-test-ppc
```
Comment (lukech) [inline test/evergreen.yml]: Can we move `ppc` ahead of `test`?

File: test/evergreen.yml (hunk 1 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Yes, I hope you remember we discussed & decided to run the "ppc" jobs to 2 hours as there are only few(11) ppc hosts.

File: test/evergreen.yml (hunk 2 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Yeah, sure. I just followed the same test setting as per Jenkins.

File: test/evergreen.yml (hunk 3 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: The compilation fails if the CCAS setting is removed. 
I have retained CCAS setting and other options are set similar to  `compile wiredtiger address sanitizer`.

File: test/evergreen.yml (hunk 4 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Fixed

File: test/evergreen.yml (hunk 5 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Fixed

File: test/evergreen.yml (hunk 6 above)
Comment (raviprakashgiri29) [reply test/evergreen.yml]: Fixed

**Comment by raviprakashgiri29** [comment]
> A few comments inline. I assume after BUILD-10248 gets resolved, we don't need the 2 new tasks - just reuse the `format-stress-sanitizer-test` and `format-stress-sanitizer-smoke-test` tasks.

I have put a comment to replace the ppc-tests with `format-stress-sanitizer-test` and `format-stress-sanitizer-smoke-test` after BUILD-10248 fix.

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen.yml]: Thanks for reminding me about it.

**Pull request merged**